src/core/proj/qgscoordinatetransform_p.cpp

```diff
@@ -288,7 +288,20 @@ ProjData QgsCoordinateTransformPrivate::threadLocalProjData()
   if ( !mProjCoordinateOperation.isEmpty() )
   {
     transform.reset( proj_create( context, mProjCoordinateOperation.toUtf8().constData() ) );
-    if ( !transform || !proj_coordoperation_is_instantiable( context, transform.get() ) )
+    // Only use proj_coordoperation_is_instantiable() if PROJ networking is enabled.
+    // The reason is that proj_coordoperation_is_instantiable() in PROJ < 9.0
+    // does not work properly when a coordinate operation refers to a PROJ < 7 grid name (gtx/gsb)
+    // but the user has installed PROJ >= 7 GeoTIFF grids.
+    // Cf https://github.com/OSGeo/PROJ/pull/3025.
```

Comment:
I'm a little confused here -- did this behaviour change somewhere after proj 6? Does proj_create return a nullptr now if the networking is disabled and the grids are not available? 

I can't recall the exact circumstances for this check when it was originally written, but it predated the proj networking so at that stage at least proj_create was returning a non-nullptr for proj_create, yet was returning a non-instantiable operation...

Reply:
> Does proj_create return a nullptr now if the networking is disabled and the grids are not available?

yes

> at least proj_create was returning a non-nullptr for proj_create, yet was returning a non-instantiable operation...

I can't reproduce that with

```
#include <proj.h>
#include <stdio.h>

int main()
{
    printf("is_null=%d\n", proj_create(NULL, "+proj=hgridshift +grids=i_dont_exist") == NULL);
    return 0;
}
```
It returns is_null=1 with PROJ 6.3.1 and master

Reply:
Ahhh.... I suspect it was probably a result of copy-and-paste from the code at line 356 or 418 which uses proj_list_get over all candidate operations and then tests instantiability for each in turn. 

I'm guessing QgsDatumTransform::transformDetailsFromPj will also need updating, since that also uses proj_coordoperation_is_instantiable ?

Reply:
> I'm guessing QgsDatumTransform::transformDetailsFromPj will also need updating, since that also uses proj_coordoperation_is_instantiable ?

Actually no, that seems fine.